enderio-conduits/src/main/java/com/enderio/conduits/common/items/ConduitProbeItem.java

```diff
@@ -0,0 +1,217 @@
+
+        if (block instanceof ConduitBundleBlockEntity conduit) {
+            if (pContext.getLevel().isClientSide()) return InteractionResult.SUCCESS;
+
+            var conduitConnection = conduit.getShape().getConnectionFromHit(pContext.getClickedPos(), pContext.getHitResult());
+            if (conduitConnection != null) {
+                Direction face = conduitConnection.getFirst();
+                switch (getState(stack)) {
+                    case COPY_PASTE -> {
+                        if (pContext.isSecondaryUseActive()) {
+                            handleCopy(conduit, face, stack, player);
+                        } else {
+                            handlePaste(conduit, face, stack, player);
+                        }
+                    }
+                    case PROBE -> {
+                        player.sendSystemMessage(Component.literal("This feature isn't implemented yet.").withStyle(ChatFormatting.RED));
+                    }
+                }
+                return InteractionResult.SUCCESS;
+            }
+        }
+        return super.onItemUseFirst(stack, pContext);
+    }
+
+    private static final Set<String> SKIP_FIELDS = Set.of("DEFAULT", "CODEC", "STREAM_CODEC", "TYPE");
+
+    private void handleCopy(ConduitBundleBlockEntity conduitBlock, Direction face, ItemStack itemStack, Player player) {
+        ProbeConfigData configData = new ProbeConfigData(new HashMap<>());
+
+        var conduits = conduitBlock.getConduits();
+        conduits.forEach(conduit -> {
+            ConnectionConfig connectionConfig = conduitBlock.getConnectionConfig(conduit, face);
+            if (connectionConfig != null) {
+                ResourceLocation conduitKey = ResourceLocation.parse(conduit.getRegisteredName());
+                var ops = net.minecraft.resources.RegistryOps.create(
+                    net.minecraft.nbt.NbtOps.INSTANCE,
+                    conduitBlock.getLevel().registryAccess()
+                );
+                var nbt = ConnectionConfig.GENERIC_CODEC
+                    .encodeStart(ops, connectionConfig)
+                    .result();
+                nbt.flatMap(tag -> ConnectionConfig.GENERIC_CODEC
+                        .parse(ops, tag)
+                        .result()
+                )
+                .ifPresent(cloned -> configData.conduitData().put(conduitKey, cloned));
+            }
+        });
+        itemStack.set(ConduitComponents.PROBE_CONFIG, configData);
+
+        if (!configData.conduitData().isEmpty()) {
+            MutableComponent message = Component.empty();
+            configData.conduitData().forEach((conduitKey, connectionConfig) -> {
+                String conduitName = conduitKeyToDisplayName(conduitKey);
+                message.append(Component.literal("\n" + conduitName + ":\n").withStyle(ChatFormatting.UNDERLINE));
+
+                for (var field : connectionConfig.getClass().getDeclaredFields()) {
```

Comment:
Just to make this method more readable, perhaps we can extract the string building out to its own method?